packages/@aws-cdk-containers/ecs-service-extensions/lib/extensions/scale-on-cpu-utilization.ts

```diff
@@ -122,19 +130,4 @@ export class ScaleOnCpuUtilization extends ServiceExtension {
       desiredCount: this.initialTaskCount,
     } as ServiceBuild;
   }
-
-  // This hook utilizes the resulting service construct
-  // once it is created.
-  public useService(service: ecs.Ec2Service | ecs.FargateService) {
-    const scalingTarget = service.autoScaleTaskCount({
-      minCapacity: this.minTaskCount,
-      maxCapacity: this.maxTaskCount,
-    });
-
-    scalingTarget.scaleOnCpuUtilization(`${this.parentService.id}-target-cpu-utilization-${this.targetCpuUtilization}`, {
-      targetUtilizationPercent: this.targetCpuUtilization,
-      scaleInCooldown: this.scaleInCooldown,
-      scaleOutCooldown: this.scaleOutCooldown,
-    });
-  }
```

Comment:
Oh I see you refactored this and moved it into the Service constructor. That is breaking the extensions pattern of the module, which is probably not the right way to do this.  Is there a way to keep this logic here?

Reply:
Ahh I see what you mean! The refactoring made it more complex then it needed to be. I have reverted the code to how it was before and just updated the `useService()` method to throw an error. This will make sure the user is either explicitly using the new, recommended way of configuring autoscaling or using the deprecated way but not both.